liquidity-pool-changes/src/agent.js

```diff
@@ -0,0 +1,193 @@
+  } catch {
+    return undefined;
+  }
+  const denominator = new BigNumber(10).pow(decimals);
+
+  // multiply by the price and divide out decimal places
+  return amountBN.times(tokenPrice).div(denominator);
+}
+
+function provideHandleBlock(data) {
+  return async function handleBlock() {
+    const { erc20Abi, provider, liquidityThresholdPercentChange } = data;
+
+    // make sure that data is initialized first
+    if (!liquidityThresholdPercentChange) throw new Error('handleBlock called before initialization');
+
+    const findings = [];
+
+    const poolContract = utils.getContract('Usdc/Eth', provider);
+
+    let token0Address;
+    let token1Address;
+    try {
+      token0Address = await poolContract.token0();
+      token1Address = await poolContract.token1();
+    } catch {
+      // asume its not a Uniswap V3 pool if contract calls fail
+      return findings;
+    }
+
+    const token0Contract = new ethers.Contract(
+      token0Address,
+      erc20Abi,
+      provider,
+    );
+    const token1Contract = new ethers.Contract(
+      token1Address,
+      erc20Abi,
+      provider,
+    );
+
+    // get the # amount of each tokens in the liquidity pool
+    const tokenBalance0 = await token0Contract.balanceOf(poolContract.address);
+    const tokenBalance1 = await token1Contract.balanceOf(poolContract.address);
+
+    // convert # amount of each tokens in the liquidity pool from ethers BigNumber to BigNumber.js
+    const tokenBalance0BN = new BigNumber(tokenBalance0.toString());
+    const tokenBalance1BN = new BigNumber(tokenBalance1.toString());
+
+    // this returns the token prices in usd of each token (without accounting for decimals)
+    let tokenPrices;
+    try {
+      tokenPrices = await getTokenPrices(token0Address, token1Address);
+    } catch (error) {
+      // if coingecko call fails
+      return findings;
+    }
+
+    // multiply the amount of tokens in the pool by usd value, adjusting for decimals
+    let token0ValueUSD;
+    let token1ValueUSD;
+    let currentLiquidity;
+
+    if (tokenBalance0BN.gt(0)) {
+      token0ValueUSD = await getTokenUSDValue(
+        tokenBalance0BN,
+        tokenPrices.token0Price,
+        token0Contract,
+      );
+    }
+
+    if (tokenBalance1BN.gt(0)) {
+      token1ValueUSD = await getTokenUSDValue(
+        tokenBalance1BN,
+        tokenPrices.token1Price,
+        token1Contract,
+      );
+    }
+
+    // calculate the total liquidity value of the pool at current block
+    if (token0ValueUSD === undefined) {
+      token0ValueUSD = new BigNumber(0);
+    }
+
+    if (token1ValueUSD === undefined) {
+      token1ValueUSD = new BigNumber(0);
+    }
+
+    // eslint-disable-next-line prefer-const
+    currentLiquidity = token0ValueUSD.plus(token1ValueUSD);
+
+    // should only be true the first time this handler is ran
+    if (data.previousLiquidity === undefined) {
+      data.previousLiquidity = currentLiquidity;
+      return findings;
```

Comment:
nit: an actual finding should never be returned at this step it makes sense to just return an empty array for clarity:
```suggestion
      return [];
```